cmd/deploy.go

```diff
@@ -107,234 +102,50 @@ func deployFn(_ *cobra.Command, _ []string) error {
 		clab.WithDebug(debug),
 	}
 
-	c, err := clab.NewContainerLab(opts...)
-	if err != nil {
-		return err
-	}
-
-	err = c.ResolveLinks()
-	if err != nil {
-		return err
-	}
-
-	c.SetClabIntfsEnvVar()
-
-	setFlags(c.Config)
-	log.Debugf("lab Conf: %+v", c.Config)
-
-	// dispatch a version check that will run in background
-	vCh := getLatestClabVersion(ctx)
-
-	if reconfigure {
-		_ = destroyLab(ctx, c)
-		log.Infof("Removing %s directory...", c.TopoPaths.TopologyLabDir())
-		if err := os.RemoveAll(c.TopoPaths.TopologyLabDir()); err != nil {
-			return err
-		}
-	}
-
-	// create management network or use existing one
-	if err = c.CreateNetwork(ctx); err != nil {
-		return err
-	}
-
-	err = links.SetMgmtNetUnderlayingBridge(c.Config.Mgmt.Bridge)
-	if err != nil {
-		return err
-	}
-
-	if err = c.CheckTopologyDefinition(ctx); err != nil {
-		return err
-	}
-
-	if err = c.LoadKernelModules(); err != nil {
-		return err
-	}
-
-	log.Info("Creating lab directory: ", c.TopoPaths.TopologyLabDir())
-	utils.CreateDirectory(c.TopoPaths.TopologyLabDir(), 0755)
-	// adjust ACL for Labdir such that SUDO_UID Users will
-	// also have access to lab directory files
-	err = utils.AdjustFileACLs(c.TopoPaths.TopologyLabDir())
-	if err != nil {
-		log.Infof("unable to adjust Labdir file ACLs: %v", err)
-	}
-
-	// create an empty ansible inventory file that will get populated later
-	// we create it here first, so that bind mounts of ansible-inventory.yml file could work
-	ansibleInvFPath := c.TopoPaths.AnsibleInventoryFileAbsPath()
-	_, err = os.Create(ansibleInvFPath)
-	if err != nil {
-		return err
-	}
-
-	// in an similar fashion, create an empty topology data file
-	topoDataFPath := c.TopoPaths.TopoExportFile()
-	topoDataF, err := os.Create(topoDataFPath)
-	if err != nil {
-		return err
+	// process optional settings
+	if name != "" {
+		opts = append(opts, clab.WithLabName(name))
 	}
-
-	if err := certificateAuthoritySetup(c); err != nil {
-		return err
+	if mgmtNetName != "" {
+		opts = append(opts, clab.WithManagementNetworkName(mgmtNetName))
 	}
-
-	c.SSHPubKeys, err = c.RetrieveSSHPubKeys()
-	if err != nil {
-		log.Warn(err)
+	if v4 := mgmtIPv4Subnet.String(); v4 != "<nil>" {
+		opts = append(opts, clab.WithManagementIpv4Subnet(v4))
```

Comment:
maybe there is a better way to check if net.IP is nil?

Reply:
It's been like this for ages (https://github.com/srl-labs/containerlab/blob/main/cmd/deploy.go#L367-L372). We can check if there is another way, but lets do that seperately.